sdcclient/_secure.py

```diff
@@ -675,7 +768,152 @@ def delete_policy_id(self, id):
+
+        **Success Return Value**
+            A JSON representation of the compliance task.
+        '''
+        task = {
+            "id": None,
+            "name": name,
+            "moduleName": module_name,
+            "enabled": enabled,
+            "scope": scope,
+            "schedule": schedule
+        }
+        res = requests.post(self.url + '/api/complianceTasks', data=json.dumps(task), headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+    def list_compliance_tasks(self):
+        '''**Description**
+            Get the list of all compliance tasks.
+
+        **Arguments**
+            - None
+
+        **Success Return Value**
+            A JSON list with the representation of each compliance task.
+        '''
+        res = requests.get(self.url + '/api/complianceTasks', headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+    def get_compliance_task(self, id):
+        '''**Description**
+            Get a compliance task.
+
+        **Arguments**
+            - id: the id of the compliance task to get.
+
+        **Success Return Value**
+            A JSON representation of the compliance task.
+        '''
+        res = requests.get(self.url + '/api/complianceTasks/{}'.format(id), headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+    def update_compliance_task(self, id, name=None, module_name=None, schedule=None, scope=None, enabled=None):
+        '''**Description**
+            Update an existing compliance task.
+
+        **Arguments**
+            - id: the id of the compliance task to be updated.
+            - name: The name of the task e.g. “Check Docker Compliance”.
+            - module_name: The name of the module that implements this task. Separate from task name in case you want to use the same module to run separate tasks with different scopes or schedules. [ 'docker-bench-security', 'kube-bench' ]
+            - schedule: The frequency at which this task should run. Expressed as an `ISO 8601 Duration <https://en.wikipedia.org/wiki/ISO_8601#Durations>`_
+            - scope: The agent will only run the task on hosts matching this scope or on hosts where containers match this scope.
+            - enabled: Whether this task should actually run as defined by its schedule.
+
+        **Success Return Value**
+            A JSON representation of the compliance task.
+        '''
+        ok, res = self.get_compliance_task(id)
+        if not ok:
+            return ok, res
+
+        task = res
+        if name is not None:
+            task["name"] = name
+        if module_name is not None:
+            task["moduleName"] = module_name
+        if schedule is not None:
+            task["schedule"] = schedule
+        if scope is not None:
+            task["scope"] = scope
+        if enabled is not None:
+            task["enabled"] = enabled
```

Comment:
I've seen this a couple of times at the code. Would this be worthy?

```python
options = {
    'name': name,
    'moduleName': module_name,
...
}

for key, value in options.items():
    if value is not None:
        task[key] = value
```

Reply:
I'm not sure how readable are dict completions, but this is a try: https://github.com/draios/python-sdc-client/pull/77/commits/33e82b255035fa514ada6293c4556d84abedc880

Reply:
Yeah, totally right. Wasn't sure at all, but wanted to know your opinion ;-)